src/contexts/ThemeProvider.jsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, useState, useContext } from "react";
+
+import { ThemeProvider as MuiThemeProvider, createTheme } from "@mui/material/styles";
+import { node } from "prop-types";
+
+import { theme } from "~/constants/theme";
+
+const ThemeContext = createContext();
+
+const HolyThemeProvider = ({ children }) => {
+  const [themeSetting, setThemeSetting] = useState({ ...theme });
```

Comment:
no need to destructure theme  as its a nested object so it with not properly make a copy it will only make a shallow copy